presentation/src/main/java/com/yapp/bol/presentation/view/group/search/GroupSearchFragment.kt

```diff
@@ -0,0 +1,115 @@
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+
+@AndroidEntryPoint
+class GroupSearchFragment : BaseFragment<FragmentGroupSearchBinding>(R.layout.fragment_group_search) {
+
+    private val viewModel: GroupSearchViewModel by activityViewModels()
+
+    override fun onViewCreatedAction() {
+        super.onViewCreatedAction()
+        binding.lifecycleOwner = this@GroupSearchFragment
+        binding.viewGroupListLoading.root.visibility = View.GONE
+        bindView()
+    }
+
+    private fun bindView() {
+        val adapter = GroupListAdapter()
+        initPaging(adapter)
+        binding.initSearchView(adapter)
+        adapter.addOnPagesUpdatedListener {
+            binding.viewGroupListLoading.root.visibility = View.GONE
+        }
+    }
+
+    private fun initPaging(adapter: GroupListAdapter) {
+        val concatAdapter = adapter.withLoadStateAdapters(
+            header = GroupListLoadStateAdapter { adapter.retry() },
+            footer = GroupListLoadStateAdapter { adapter.retry() }
+        )
+        binding.rvGroupList.adapter = concatAdapter
+    }
+
+    // 상단 search view 초기화 관련 작업
+    private fun FragmentGroupSearchBinding.initSearchView(adapter: GroupListAdapter) {
+        binding.initEditText(adapter)
+
+        val editText = viewGroupSearch.etGroupSearch
+        val rightDrawableBtn = viewGroupSearch.btnSearchRight
+
+        rightDrawableBtn.setOnClickListener {
+            if (!editText.text.isNullOrEmpty() || !editText.text.isNullOrBlank()) {
+                editText.text.clear()
+            }
+        }
+
+        editText.loseFocusOnAction(EditorInfo.IME_ACTION_SEARCH, this.root.context)
+
+        binding.viewGroupSearch.btnCreateGroup.setOnClickListener {
+            // TODO : create transition
+        }
+    }
+
+    // search view의 edittext 세팅
+    private fun FragmentGroupSearchBinding.initEditText(adapter: GroupListAdapter) {
+        viewLifecycleOwner.lifecycleScope.launch {
+            val editTextFlow = viewGroupSearch.etGroupSearch.textChangesToFlow()
+            val debounceDuration = 500L
+
+            editTextFlow
+                .onEach {
+                    val isTyping = !(it.isNullOrBlank() || it.isEmpty())
+                    viewGroupSearch.btnSearchRight.setImageButtonByState(isTyping)
+                }
+                .launchIn(this)
+
+            editTextFlow
+                .debounce(debounceDuration)
+                .onEach {
+                    adapter.searchByKeyword(it.toString())
+                }
+                .launchIn(this)
+        }
+    }
+
+    private fun GroupListAdapter.searchByKeyword(keyword: String) {
+        viewLifecycleOwner.lifecycleScope.launch {
+            submitData(PagingData.empty())
+            viewModel.searchGroup(keyword).collectLatest {
+                submitData(it)
+            }
+        }
```

Comment:
p3 : 제가 flow collect 확장함수 만들어서 요렇게 작성할 수도 있어요.
```suggestion
        viewModel.searchGroup(keyword).collectWithLifecycle(viewLifecycleOwner) {
            submitData(PagingData.empty())
            submitData(it)
        }
```
또한 adapter 에 data 를 submit 하기전 empty 를 전달하는 이유가 뭔지 궁금해요!

Reply:
이건 제가 검색할 때마다 loadStateAdapter를 사용한 로딩 화면 -> 검색 결과 리스트 한 번에 보여주도록 구현하고 싶어서 empty를 전달하였습니다!